inbox/util/concurrency.py

```diff
@@ -2,24 +2,30 @@
 import random
 
 import gevent
+from backports import ssl
+from gevent import socket
+from redis import TimeoutError
 
 from nylas.logging import get_logger
 from nylas.logging.sentry import log_uncaught_errors
 log = get_logger()
+
 BACKOFF_DELAY = 30  # seconds to wait before retrying after a failure
+TRANSIENT_NETWORK_ERRS = (socket.timeout, TimeoutError, socket.error, ssl.SSLError)
```

Comment:
Hmmm, but the other side of the SSL connection could change their certificate—so even if an SSL error is permanent given a certain certificate, it's effectively transient because we don't control that certificate. Yeah?